### ISSUE TITLE
CMake does not work

### ISSUE THREAD

**Issue by heinrich5991** [Dune-jr/teeworlds#77]
The cmake build fails due to some data files and source files not being listed.

**Comment by Dune-jr** [comment]
Yeah, it doesn't :/
CMake messes with my install and caused a few headaches, so I'll look for a way to have it list all the files in the source folder like bam does to avoid constantly having to switch.

**Comment by Dune-jr** [comment]
Didn't manage to achieve that after a while.

I see that cmake is supposed to show info of missing entries
```

      if(LIST_BUT_NOT_GLOB)
        message(AUTHOR_WARNING "Entries only present in ${VAR}: ${LIST_BUT_NOT_GLOB}")
      endif()
      set(GLOB_BUT_NOT_LIST)
      foreach(file ${GLOB_RESULT})
        if(NOT file IN_LIST FILES)
          list(APPEND GLOB_BUT_NOT_LIST ${file})
        endif()
      endforeach()
      if(GLOB_BUT_NOT_LIST)
        message(AUTHOR_WARNING "Entries only present in ${DIRECTORY}: ${GLOB_BUT_NOT_LIST}")
      endif()
```
I get none of that though :(

```
CMake Error (dev) at CMakeLists.txt:200 (message):
  DATA does not contain every file from directory datasrc
Call Stack (most recent call first):
  CMakeLists.txt:463 (set_glob)
This error is for project developers. Use -Wno-error=dev to suppress it.

```

**Comment by heinrich5991** [comment]
I recently PR'ed this on the Teeworlds repository, this happens when the lists are not alphabetically ordered.

**Comment by Dune-jr** [comment]
That's not it, the list is already alphabetically sorted, and I tested with an up-to-date branch. Only some files are missing.

My point was that there seems to be some warning code that isn't run

**Comment by Dune-jr** [comment]
Reminder to self: try `file(GLOB src/*.cpp)`

**Comment by heinrich5991** [comment]
It can work with file globbing, in fact the current CMakeLists.txt use `file(GLOB …)` to verify that the given files are all that are listed in the directory. You can probably modify it to just take that list instead of the passed one. You'll want to pass the `CONFIGURE_DEPENDS` flag to `file(GLOB …)`, otherwise it'll break when new files are added/old ones are removed.

Note that CMake discourages the use of `file(GLOB …)` to collect source files, from https://cmake.org/cmake/help/v3.15/command/file.html#glob:
>**Note:** We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The `CONFIGURE_DEPENDS` flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if `CONFIGURE_DEPENDS` works reliably, there is still a cost to perform the check on every rebuild.

**Comment by Dune-jr** [comment]
Fixed by 61da99b5fc287ffa89c4327d965dfdbb37cf5fce, thanks
